pkg/core/mempool/mem_pool.go

```diff
@@ -403,6 +420,27 @@ func (mp *Pool) TryGetValue(hash util.Uint256) (*transaction.Transaction, bool)
 	return nil, false
 }
 
+// TryGetData returns data associated with the specified transaction if it exists in the memory pool.
+func (mp *Pool) TryGetData(hash util.Uint256) (interface{}, bool) {
+	mp.lock.RLock()
+	defer mp.lock.RUnlock()
+	if tx, ok := mp.verifiedMap[hash]; ok {
+		itm := item{txn: tx}
+		n := sort.Search(len(mp.verifiedTxes), func(n int) bool {
+			return itm.CompareTo(mp.verifiedTxes[n]) >= 0
+		})
+		if n < len(mp.verifiedTxes) {
+			for i := n; i < len(mp.verifiedTxes); i++ { // items may have equal priority, so `n` is the left bound of the items which are as prioritized as the desired `itm`.
```

Comment:
You can break out of the loop if `itm.CompareTo() != 0`.

Reply:
I thought about it, but there might be a case when a lot of items are iterated before this condition is satisfied. The other way to stop the cycle is to search the right bound of items which are as prioritized as the desired one by using `sort.Search` with `itm.CompareTo(mp.verifiedTxes[n]) > 0` condition. Which is more preferable?

Reply:
> a case when a lot of items are iterated before this condition is satisfied

But it's `<= len(mp.verifiedTxes)` anyway. There are from 0 to `len(mp.verifiedTxes) - n` transactions satisfying `itm.CompareTo() == 0`.